pkg/ottl/expression.go

```diff
@@ -139,36 +139,123 @@ type StringGetter[K any] interface {
 	Get(ctx context.Context, tCtx K) (string, error)
 }
 
+type StandardStringGetter[K any] struct {
+	Getter func(ctx context.Context, tCtx K) (interface{}, error)
+}
+
+func (g StandardStringGetter[K]) Get(ctx context.Context, tCtx K) (string, error) {
+	val, err := g.Getter(ctx, tCtx)
+	if err != nil {
+		return "", err
+	}
+	if val == nil {
+		return "", fmt.Errorf("expected string but got nil")
+	}
+	switch v := val.(type) {
+	case string:
+		return v, nil
+	case pcommon.Value:
+		if v.Type() == pcommon.ValueTypeStr {
+			return v.Str(), nil
+		}
+		return "", fmt.Errorf("expected string but got %v", v.Type())
```

Comment:
Would it make more sense to say something like "expected pcommon.ValueTypeStr" here? Technically we're not expecting a string.

Reply:
I stuck with `string` because ultimately the Getter wants a string and we're being generous letting a pcommon.Value that holds a string value count as a string.  Since we're adding this functionality to obfuscate the difference between a string and a `pcommon.Value` I stuck with `string` for the message.

Reply:
That's a fair point. Let's keep it as-is.